tests/test_config.py

```diff
@@ -0,0 +1,20 @@
+"""Pytest for testing kwargs in clients"""
+
+import pytest
+from otelib import OTEClient
+
+
+@pytest.mark.parametrize(
+    "backend,config",
+    [
+        ("python", {"headers": {"foo": "bar"}}),
+        ("http://localhost:8080", {"foo": "bar"}),
+    ],
+)
+def test_warning(backend, config):
+    """Pytest for unloaded kwargs warning."""
+    message = f"The given configuration option(s) for {tuple(config)} is/are ignored."
+    with pytest.warns(UserWarning) as record:
```

Comment:
@CasperWA, due to any reason I do not understand, the `UserWarning` is not properly caputered by using the `match`-kwarg in the `pytest.warns` method. 

Therefore I had to use the `record` object in order to validate the messages. 

If find a solution why the proper handling of the `pytest.warns` fails here, feel free to update here :)